extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -312,11 +312,11 @@ export class AmpAdExit extends AMP.BaseElement {
         this.defaultFilters_.splice(
           0,
           1,
-          createFilter(
+          /** @type {!./filters/filter.Filter} */ (createFilter(
```

Comment:
Have createFilter specify its return type?

Reply:
Already [specified](https://github.com/rsimha/amphtml/blob/2019-07-26-ReturnAnnotation/extensions/amp-ad-exit/0.1/filters/factory.js#L22-L46). The cast is necessary because `undefined` is a possible return type.

Reply:
Reading `createFilter` it sounds like `undefined` happens if it's given an unknown spec type.  But `amp-ad-exit` is the only extension using `createFilter` and all three of its `createFilter` calls shouldn't result in `undefined`.  Giving an unknown spec to `createFilter` looks like a programming error, so how about modifying `createFilter` to throw in that case, and removing the possibility of it returning `undefined`?

Reply:
Good suggestion. Out of scope for this PR.